src/workerd/api/actor-state.c++

```diff
@@ -676,8 +676,18 @@ jsg::JsRef<jsg::JsValue> DurableObjectStorage::transactionSync(
       sqlite.run(SqliteDatabase::TRUSTED, kj::str("RELEASE _cf_sync_savepoint_", depth));
       return kj::mv(result);
     }, [&](jsg::Value exception) -> jsg::JsRef<jsg::JsValue> {
-      sqlite.run(SqliteDatabase::TRUSTED, kj::str("ROLLBACK TO _cf_sync_savepoint_", depth));
-      sqlite.run(SqliteDatabase::TRUSTED, kj::str("RELEASE _cf_sync_savepoint_", depth));
+      if (sqlite.observedCriticalError()) {
+        // The database has observed a critical error that forced an automatic rollback.  SQLite
+        // docs recommend attempting an explicit rollback, even though it is expected to fail due
```

Comment:
I think the recommendation in the sqlite docs only applies if the application is _not_ using `sqlite3_get_autocommit()` to determine explicitly whether a rollback already occurred. We do that before flagging a critical error, so I think we can skip trying to rollback here -- it will definitely always fail.

Reply:
Ah, good to know, thanks... I've updated the code to skip the transaction rollback/release attempts.